client.go

```diff
@@ -182,6 +182,12 @@ type Client interface {
 	// To scan the next set of rows, modify the scanOp to provide
 	// the string returned as an Offset()
 	ScanEverything(ctx context.Context, scanOp *ScanOp) ([]DomainObject, string, error)
+
+	// Shutdown gracefully shutsdown the client, cleaning up any resources it may have
+	// allocated during it's usage. Shutdown should be called whenever the client
```

Comment:
Two nits: 
 - shutsdown -> shuts down
 - "it's" == "it is", the possessive is "its"